components/hab/src/main.rs

```diff
@@ -784,8 +784,12 @@ fn sub_pkg_provides(m: &ArgMatches<'_>) -> Result<()> {
 fn sub_pkg_search(m: &ArgMatches<'_>) -> Result<()> {
     let url = bldr_url_from_matches(&m)?;
     let search_term = m.value_of("SEARCH_TERM").unwrap(); // Required via clap
+    let limit = m.value_of("LIMIT").unwrap().parse().unwrap(); // Required via clap
```

Comment:
I learned my lesson awhile back about using `expect` even when failure should be impossible, since it makes zeroing in on the issue in release builds *way* easier. I'd suggest changing both this and the above line.
```suggestion
    let limit = m.value_of("LIMIT")
                 .expect("required opt LIMIT")
                 .parse()
                 .expect("valid LIMIT");
```

Reply:
👍